ibis/expr/operations/logical.py

```diff
@@ -157,3 +162,39 @@ class Where(Value):
 
     output_dtype = rlz.dtype_like("true_expr")
     output_shape = rlz.shape_like("bool_expr")
+
+
+class _AnyBase(Value):
+    # Depending on the kind of input boolean array, the result might either be
+    # array-like (an existence-type predicate) or scalar (a reduction)
+    arg = rlz.column(rlz.boolean)
+
+    output_dtype = dt.boolean
+
+    @property
+    def _reduction(self):
```

Comment:
I think we should define two seperate operations instead, one is a reduction the other is not.

Reply:
We could construct them from the expression's corresponding method based on `arg`'s type.

Reply:
Agree, but not as part of this PR. This PR is only about moving the transformation out of the compiler to expression construction.

Reply:
Sure, could you create an issue?

Reply:
Done: https://github.com/ibis-project/ibis/issues/3925.